artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPConnectionContext.java

```diff
@@ -143,13 +144,26 @@ public SASLResult getSASLResult() {
    }
 
    public void inputBuffer(ByteBuf buffer) {
-      if (log.isTraceEnabled()) {
-         ByteUtil.debugFrame(log, "Buffer Received ", buffer);
+      if (PN_TRACE_FRM) {
+         debugFrame(buffer);
       }
 
       handler.inputBuffer(buffer);
    }
 
+   private static void debugFrame(ByteBuf byteIn) {
+      int location = byteIn.readerIndex();
+      // debugging
+      byte[] frame = new byte[byteIn.writerIndex()];
+      byteIn.readBytes(frame);
+
+      if (PN_TRACE_FRM) {
```

Comment:
This was checked already before calling the method. Ignoring that it would probably also be better done around the whole contents of the method to avoid creating the byte array but then not using it and then needing to readjust the buffer index later.